kernel/run.py

```diff
@@ -504,8 +511,9 @@ def pmem_paths(args):
             print("STDERR: {}".format(execution.stderr.decode('utf-8')))
 
     # If the test creates a fake pmem path; remove it.
-    if os.path.isfile(pmem_test_path):
-        os.remove(pmem_test_path)
+    for path in pmem_paths(args):
```

Comment:
can we use `with` block and [NamedTemporaryFile](https://docs.python.org/2/library/tempfile.html#tempfile.NamedTemporaryFile) so this gets cleaned up even if something fails randomly further up?

Reply:
Thanks, I didn't know that something like this existed. Updated; could you please check if the changes rectify your concerns.